go/parquet/internal/encoding/fixed_len_byte_array_encoder.go

```diff
@@ -75,6 +75,54 @@ func (PlainFixedLenByteArrayEncoder) Type() parquet.Type {
 	return parquet.Types.FixedLenByteArray
 }
 
+// ByteStreamSplitFixedLenByteArrayEncoder writes the raw bytes of the byte array
+// always writing typeLength bytes for each value. (TODO)
+type ByteStreamSplitFixedLenByteArrayEncoder struct {
+	encoder
+
+	bitSetReader bitutils.SetBitRunReader
+}
+
+// Put writes the provided values to the encoder
+func (enc *ByteStreamSplitFixedLenByteArrayEncoder) Put(in []parquet.FixedLenByteArray) {
+	bytesNeeded := len(in) * enc.typeLen
+	enc.sink.Reserve(bytesNeeded)
+	for offset := 0; offset < enc.typeLen; offset++ {
+		for _, val := range in {
+			if len(val) != enc.typeLen {
+				panic(fmt.Sprintf("invalid element of size %d in FixedLenByteArray of size %d", len(val), enc.typeLen))
+			}
```

Comment:
There's gotta be a way to not make this check on every iteration since we'll make this comparison K * N times and we only need to make this comparison on the first iteration of the outer loop, rather than every iteration

Reply:
Yes definitely, that was at the top of my list to cleanup. Just wanted to get my WIP up as a draft for now.